From f2216cab3558528f8a8c8836addaf936fc61c4dc Mon Sep 17 00:00:00 2001
From: Nathan Chancellor <nathan@kernel.org>
Date: Fri, 14 May 2021 10:36:23 -0700
Subject: [PATCH 22/28] DO-NOT-UPSTREAM: Various fixes/hacks/warning hiding

* drivers/input/joystick/analog.c: Avoid this warning on certain
  architectures.

* lib/{Kconfig.debug,Makefile}: Disable -Wframe-larger-than=. Getting
  all of these fixed is a long term project.

* scripts/checksyscalls.sh: I do not care for this warning.

Link: https://github.com/ClangBuiltLinux/linux/labels/-Wframe-larger-than%3D
Signed-off-by: Nathan Chancellor <nathan@kernel.org>
---
 drivers/input/joystick/analog.c          | 1 -
 drivers/net/ethernet/sfc/Makefile        | 2 ++
 drivers/net/ethernet/sfc/falcon/Makefile | 2 ++
 lib/Kconfig.debug                        | 3 ++-
 lib/Makefile                             | 1 -
 scripts/checksyscalls.sh                 | 2 +-
 6 files changed, 7 insertions(+), 4 deletions(-)

diff --git a/drivers/input/joystick/analog.c b/drivers/input/joystick/analog.c
index f798922a4598..dea166758af1 100644
--- a/drivers/input/joystick/analog.c
+++ b/drivers/input/joystick/analog.c
@@ -157,7 +157,6 @@ static unsigned long analog_faketime = 0;
 #define GET_TIME(x)     do { x = analog_faketime++; } while(0)
 #define DELTA(x,y)	((y)-(x))
 #define TIME_NAME	"Unreliable"
-#warning Precise timer not defined for this architecture.
 #endif
 
 static inline u64 get_time(void)
diff --git a/drivers/net/ethernet/sfc/Makefile b/drivers/net/ethernet/sfc/Makefile
index 8bd01c429f91..4331ed72bd69 100644
--- a/drivers/net/ethernet/sfc/Makefile
+++ b/drivers/net/ethernet/sfc/Makefile
@@ -13,3 +13,5 @@ sfc-$(CONFIG_SFC_SRIOV)	+= sriov.o siena_sriov.o ef10_sriov.o
 obj-$(CONFIG_SFC)	+= sfc.o
 
 obj-$(CONFIG_SFC_FALCON) += falcon/
+# https://github.com/ClangBuiltLinux/linux/issues/1439
+CFLAGS_farch.o := $(call cc-disable-warning, shift-count-negative)
diff --git a/drivers/net/ethernet/sfc/falcon/Makefile b/drivers/net/ethernet/sfc/falcon/Makefile
index 39448e5b7f1e..de1b626e12b0 100644
--- a/drivers/net/ethernet/sfc/falcon/Makefile
+++ b/drivers/net/ethernet/sfc/falcon/Makefile
@@ -5,3 +5,5 @@ sfc-falcon-y		+= efx.o nic.o farch.o falcon.o tx.o rx.o selftest.o \
 
 sfc-falcon-$(CONFIG_SFC_FALCON_MTD)	+= mtd.o
 obj-$(CONFIG_SFC_FALCON)		+= sfc-falcon.o
+# https://github.com/ClangBuiltLinux/linux/issues/1439
+CFLAGS_farch.o := $(call cc-disable-warning, shift-count-negative)
diff --git a/lib/Kconfig.debug b/lib/Kconfig.debug
index 488d0bab547e..012d58428318 100644
--- a/lib/Kconfig.debug
+++ b/lib/Kconfig.debug
@@ -343,8 +343,9 @@ config GDB_SCRIPTS
 endif # DEBUG_INFO
 
 config FRAME_WARN
-	int "Warn for stack frames larger than"
+	int "Warn for stack frames larger than" if !CC_IS_CLANG
 	range 0 8192
+	default 0 if CC_IS_CLANG
 	default 2048 if GCC_PLUGIN_LATENT_ENTROPY
 	default 1536 if (!64BIT && PARISC)
 	default 1024 if (!64BIT && !PARISC)
diff --git a/lib/Makefile b/lib/Makefile
index 2cfd33917ad5..0014b2ac911c 100644
--- a/lib/Makefile
+++ b/lib/Makefile
@@ -351,7 +351,6 @@ obj-$(CONFIG_OBJAGG) += objagg.o
 obj-$(CONFIG_PLDMFW) += pldmfw/
 
 # KUnit tests
-CFLAGS_bitfield_kunit.o := $(call cc-option,-Wframe-larger-than=10240)
 obj-$(CONFIG_BITFIELD_KUNIT) += bitfield_kunit.o
 obj-$(CONFIG_LIST_KUNIT_TEST) += list-test.o
 obj-$(CONFIG_LINEAR_RANGES_TEST) += test_linear_ranges.o
diff --git a/scripts/checksyscalls.sh b/scripts/checksyscalls.sh
index b7609958ee36..5f42da1b0a1a 100755
--- a/scripts/checksyscalls.sh
+++ b/scripts/checksyscalls.sh
@@ -259,7 +259,7 @@ EOF
 syscall_list() {
     grep '^[0-9]' "$1" | sort -n |
 	while read nr abi name entry ; do
-		echo "#if !defined(__NR_${name}) && !defined(__IGNORE_${name})"
+		echo "#if 0"
 		echo "#warning syscall ${name} not implemented"
 		echo "#endif"
 	done
-- 
2.33.0

